homeassistant/components/light/xiaomi_miio.py

```diff
@@ -435,13 +496,30 @@ def async_update(self):
             })
 
         except DeviceException as ex:
-            self._state = None
+            self._available = False
             _LOGGER.error("Got exception while fetching the state: %s", ex)
 
+    @staticmethod
+    def translate(value, left_min, left_max, right_min, right_max):
+        """Map a value from left span to right span."""
+        left_span = left_max - left_min
+        right_span = right_max - right_min
+        value_scaled = float(value - left_min) / float(left_span)
+        return int(right_min + (value_scaled * right_span))
+
 
-class XiaomiPhilipsCeilingLamp(XiaomiPhilipsLightBall, Light):
+class XiaomiPhilipsCeilingLamp(XiaomiPhilipsBulb):
     """Representation of a Xiaomi Philips Ceiling Lamp."""
 
+    def __init__(self, name, light, model, unique_id):
+        """Initialize the light device."""
+        XiaomiPhilipsBulb.__init__(self, name, light, model, unique_id)
```

Comment:
The init calls should probably be done on `super()` (as that will keep it working even if the name of the super class changes.